### PR TITLE
GC-888 - Added login_user and missing doc strings to application_users.

### PR DESCRIPTION
**Pull request by ramonPaulAlvarez**

### CODE REVIEW

File: src/evrythng/entities/application_users.py [hunk 1]
```diff
@@ -74,22 +78,33 @@ def authenticate_facebook_user(expires, token, api_key=None):
...
     url = '/users/{}'.format(user_id)
     return utils.request('DELETE', url, api_key=api_key)
 
 
 def logout_user(api_key=None):
+    """Logout an Application User."""
     return utils.request('POST', '/auth/all/logout', api_key=api_key)
 
 
+def login_user(email, password, api_key):
```
Comment (LyleScott) [inline src/evrythng/entities/application_users.py]: All other api_key arguments in all other functions have a default of None... is there a reason this one doesn't?

Comment (LyleScott) [inline src/evrythng/entities/application_users.py]: refreshed my memory. looks like it should have a default argument of None

Comment (ramonPaulAlvarez) [inline src/evrythng/entities/application_users.py]: Simply that it's required and it's the App Key.  If we don't force them to provide a key we would resort to the Operator Key and that isn't what we should use.

**Comment by LyleScott** [comment]
CR+1

**Pull request closed without merging**